orchestrator/modules/memory/memory_tool.py

```
@@ -0,0 +1,260 @@
+        if deleter is None:
+            logger.info(
+                "[memory-tool] durable store has no delete — leaving stale entry "
+                "for %s (§8-Q6: substrate is Gerard's call)", path,
+            )
+            return
+        try:
+            results = await svc.search_long_term(workspace_id=workspace_id, query=path, limit=25)
+            for entry in results or []:
+                if self._entry_path(entry) == path and isinstance(entry, dict) and entry.get("id"):
+                    await deleter(workspace_id=workspace_id, memory_id=entry["id"])
+        except Exception as exc:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)